### PR TITLE
OCP-24714 - NAD default gateway is updated when SriovNetwork ipam is …

### PR DESCRIPTION
**Pull request by mmirecki**
…changed

### CODE REVIEW

**Comment by fedepaol** [comment]
@mmirecki you should rebase this

File: test/conformance/tests/sriov_operator.go [hunk 1]
```diff
@@ -913,9 +921,76 @@ var _ = Describe("[sriov] operator", func() {
...
+					"rangeStart": "10.11.11.171",
+					"rangeEnd": "10.11.11.181",
+					"routes": [{
+					  "dst": "0.0.0.0/0"
+					}],
+					"gateway": "10.11.11.1"
+				  }`
+				err = network.CreateSriovNetwork(clients, sriovDevice, sriovNetworkName, namespaces.Test, operatorNamespace, resourceName, ipam)
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:946]: nit: new line here please

File: test/conformance/tests/sriov_operator.go [hunk 2]
```diff
@@ -913,9 +921,76 @@ var _ = Describe("[sriov] operator", func() {
...
+					netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+					return clients.Get(context.Background(), runtimeclient.ObjectKey{Name: sriovNetworkName, Namespace: namespaces.Test}, netAttDef)
+				}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+				Eventually(func() string {
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:950]: put this in the eventually above please.

File: test/conformance/tests/sriov_operator.go [hunk 3]
```diff
@@ -913,9 +921,76 @@ var _ = Describe("[sriov] operator", func() {
...
+					netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+					clients.Get(context.Background(), runtimeclient.ObjectKey{Name: sriovNetworkName, Namespace: namespaces.Test}, netAttDef)
+					nadConfig := nadConfig{}
+					json.Unmarshal([]byte(netAttDef.Spec.Config), &nadConfig)
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:954]: I think we can just use the config from type string and run a `strings.contains` for `"gateway": "10.11.11.1"`

File: test/conformance/tests/sriov_operator.go [hunk 4]
```diff
@@ -913,9 +921,76 @@ var _ = Describe("[sriov] operator", func() {
...
+		Context("NAD update", func() {
+			// 24714
+			It("NAD default gateway is updated when SriovNetwork ipam is changed", func() {
+				resourceName := "sriovnic"
+				sriovNetworkName := "sriovnetwork"
+				testNode := sriovInfos.Nodes[0]
+				sriovDevice, err := sriovInfos.FindOneSriovDevice(testNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				ipam := `{
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:934]: Can you create this as a variable for the function and use `fmt.sprintf`  with a `%s` in the gateway

File: test/conformance/tests/sriov_operator.go [hunk 5]
```diff
@@ -913,9 +921,76 @@ var _ = Describe("[sriov] operator", func() {
...
+				}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+				Eventually(func() string {
+					netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+					clients.Get(context.Background(), runtimeclient.ObjectKey{Name: sriovNetworkName, Namespace: namespaces.Test}, netAttDef)
+					nadConfig := nadConfig{}
+					json.Unmarshal([]byte(netAttDef.Spec.Config), &nadConfig)
+					return nadConfig.Ipam.Gateway
+				}, 30*time.Second, 1*time.Second).Should(Equal("10.11.11.1"))
+
+				ipam = `{
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:958]: please use the variable from above and pass `fmt.sprintf` with the new gateway

File: test/conformance/tests/sriov_operator.go [hunk 6]
```diff
@@ -913,9 +921,76 @@ var _ = Describe("[sriov] operator", func() {
...
+					"type": "host-local",
+					"subnet": "10.11.11.0/24",
+					"rangeStart": "10.11.11.171",
+					"rangeEnd": "10.11.11.181",
+					"routes": [{
+					  "dst": "0.0.0.0/0"
+					}],
+					"gateway": "10.12.11.1"
+				  }`
+				body, _ := json.Marshal([]patchBody{{
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:968]: I think there is no need for the patch you can just run update on the sriov network object

File: test/conformance/tests/sriov_operator.go (hunk 1 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:946]: added one line above

File: test/conformance/tests/sriov_operator.go (hunk 2 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:950]: Eventuallies merged

File: test/conformance/tests/sriov_operator.go (hunk 3 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:954]: Done

File: test/conformance/tests/sriov_operator.go (hunk 4 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:934]: Done

File: test/conformance/tests/sriov_operator.go (hunk 6 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go]: Done

**Pull request merged**